runtimes/evm-rollup/src/precompiles.rs

```diff
@@ -0,0 +1,95 @@
+//! Precompile set for the EVM runtime.
+//!
+//! This module defines the precompiles available to EVM contracts.
+
+use core::marker::PhantomData;
+
+use confidential_assets_evm_precompile::ConfidentialAssetsPrecompile;
+use frame_support::dispatch::{GetDispatchInfo, PostDispatchInfo};
+use pallet_evm::{
+    AddressMapping, IsPrecompileResult, Precompile, PrecompileHandle, PrecompileResult,
+    PrecompileSet,
+};
+use pallet_evm_precompile_modexp::Modexp;
+use pallet_evm_precompile_sha3fips::Sha3FIPS256;
+use pallet_evm_precompile_simple::{ECRecover, ECRecoverPublicKey, Identity, Ripemd160, Sha256};
+use sp_core::H160;
+use sp_runtime::traits::Dispatchable;
+
+/// Address for the confidential assets precompile (0x800 = 2048)
+pub const CONFIDENTIAL_ASSETS_PRECOMPILE: u64 = 2048;
+
+/// The precompile set for the EVM runtime.
+#[derive(Default)]
+pub struct FrontierPrecompiles<R>(PhantomData<R>);
+
+impl<R> FrontierPrecompiles<R>
+where
+    R: pallet_evm::Config,
+{
+    pub fn new() -> Self {
+        Self(Default::default())
+    }
+
+    /// Returns the list of addresses used by precompiles.
+    pub fn used_addresses() -> [H160; 8] {
+        [
+            hash(1),                              // ECRecover
+            hash(2),                              // Sha256
+            hash(3),                              // Ripemd160
+            hash(4),                              // Identity
+            hash(5),                              // Modexp
+            hash(1024),                           // Sha3FIPS256
+            hash(1025),                           // ECRecoverPublicKey
```

Comment:
that's a nitpick, but let's write the numbers in hex for better visibility? when they will appear in scanners and debugging tools they will be in hex